### PR TITLE
replace structure() with new_tibble() in new_grouped_df()

### PR DESCRIPTION
**Pull request by earowang**
Seems that `*` in the tibble's printing indicates row.names, which is introduced by `structure()`. `new_tibble()` (as performant as `structure()`) takes care of not having row.names.

``` r
library(dplyr, warn.conflicts = FALSE)
class(iris) <- c("tbl_xyz", "tbl_df", "tbl", "data.frame")
grped_iris <- group_by(iris, Species)
new_grouped_df(iris, groups = group_data(grped_iris))
#> # A tibble: 150 x 5
#> # Groups:   Species [3]
#>    Sepal.Length Sepal.Width Petal.Length Petal.Width Species
#>  *        <dbl>       <dbl>        <dbl>       <dbl> <fct>  
#>  1          5.1         3.5          1.4         0.2 setosa 
#>  2          4.9         3            1.4         0.2 setosa 
#>  3          4.7         3.2          1.3         0.2 setosa 
#>  4          4.6         3.1          1.5         0.2 setosa 
#>  5          5           3.6          1.4         0.2 setosa 
#>  6          5.4         3.9          1.7         0.4 setosa 
#>  7          4.6         3.4          1.4         0.3 setosa 
#>  8          5           3.4          1.5         0.2 setosa 
#>  9          4.4         2.9          1.4         0.2 setosa 
#> 10          4.9         3.1          1.5         0.1 setosa 
#> # … with 140 more rows
```

With PR, `*` is removed using `new_tibble()`:
``` r
library(dplyr, warn.conflicts = FALSE)
class(iris) <- c("tbl_xyz", "tbl_df", "tbl", "data.frame")
grped_iris <- group_by(iris, Species)
new_grouped_df(iris, groups = group_data(grped_iris))
#> # A tibble: 150 x 5
#> # Groups:   Species [3]
#>    Sepal.Length Sepal.Width Petal.Length Petal.Width Species
#>           <dbl>       <dbl>        <dbl>       <dbl> <fct>  
#>  1          5.1         3.5          1.4         0.2 setosa 
#>  2          4.9         3            1.4         0.2 setosa 
#>  3          4.7         3.2          1.3         0.2 setosa 
#>  4          4.6         3.1          1.5         0.2 setosa 
#>  5          5           3.6          1.4         0.2 setosa 
#>  6          5.4         3.9          1.7         0.4 setosa 
#>  7          4.6         3.4          1.4         0.3 setosa 
#>  8          5           3.4          1.5         0.2 setosa 
#>  9          4.4         2.9          1.4         0.2 setosa 
#> 10          4.9         3.1          1.5         0.1 setosa 
#> # … with 140 more rows
```

<sup>Created on 2019-02-11 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

### CODE REVIEW

File: R/grouped-df.r [hunk 1]
```diff
@@ -50,11 +50,12 @@ new_grouped_df <- function(x, groups, ..., class = character()) {
     is.data.frame(groups),
     tail(names(groups), 1L) == ".rows"
   )
-  structure(
+  tibble::new_tibble(
```
Comment (romainfrancois) [inline R/grouped-df.r]: Can you `@importFrom tibble new_tibble` instead?

Comment (earowang) [reply R/grouped-df.r]: done.

File: tests/testthat/test-new_grouped_df.R [hunk 2]
```diff
@@ -36,3 +36,11 @@ test_that("validate_grouped_df (#3837)", {
   attr(df, "groups") <- NULL
   expect_error(validate_grouped_df(df), "The `groups` attribute is not a data frame with its last column called `.rows`")
 })
+
+test_that("new_grouped_df has rownames (#4173)", {
```
Comment (romainfrancois) [inline tests/testthat/test-new_grouped_df.R]:
```suggestion
test_that("new_grouped_df does not have rownames (#4173)", {
```

File: tests/testthat/test-new_grouped_df.R [hunk 3]
```diff
@@ -36,3 +36,11 @@ test_that("validate_grouped_df (#3837)", {
...
+  tbl <- new_grouped_df(
+    tibble(x = rnorm(10)),
+    groups = tibble(".rows" := replicate(5, sample(1:10, replace = TRUE), simplify = FALSE))
+  )
+  expect_true(!tibble::has_rownames(tbl))
```
Comment (romainfrancois) [inline tests/testthat/test-new_grouped_df.R]:
```suggestion
  expect_false(tibble::has_rownames(tbl))
```

**Comment by romainfrancois** [comment]
Thanks.

**Pull request merged**